### PR TITLE
SLB-278: optional CMS integrations

### PR DESCRIPTION
**Pull request by pmelab**
## Description of changes

Refactor the codebase to make both Drupal and Decap optional.
To achieve this, the following changes have been made:

* Make Drupal and Decap Gatsby plugins, move CMS-specific configurations there.
* Split e2e tests in Drupal and Decap suites. Make sure both pass in each scenario.
* Make the `@custom/website` package the central entrypoint for testing, so a CMS that is not enabled, is also not tested.

## Motivation and context

Make it as easy as possible to start a project with Drupal, Decap, both or none of them.

## How has this been tested?

- [x] Manually
- [x] Unit tests
- [x] Integration tests

### CODE REVIEW

File: apps/website/gatsby-config.mjs [hunk 1]
```diff
@@ -6,97 +6,86 @@
...
+      schema_configuration: './graphqlrc.yml',
+    },
+  },
+  '@custom/cms',
+  '@custom/decap',
+];
+
+// Probe if Drupal is there and use it to source data.
+if (existsSync('./node_modules/@custom/cms/package.json')) {
+  plugins.push();
```
Comment (Leksat) [inline apps/website/gatsby-config.mjs]: Forgotten?

File: apps/cms/gatsby-config.mjs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    resolve: '@amazeelabs/gatsby-source-silverback',
+    options: {
+      schema_configuration: './graphqlrc.yml',
+      directives: autoload,
+      drupal_url: process.env.DRUPAL_INTERNAL_URL || 'http://127.0.0.1:8888',
+      drupal_external_url:
+        // File requests are proxied through netlify.
+        process.env.NETLIFY_URL || 'http://127.0.0.1:8000',
+      graphql_path: '/graphql',
+      auth_key: 'cfdb0555111c0f8924cecab028b53474',
```
Comment (Leksat) [inline apps/cms/gatsby-config.mjs:20]: We need to update the file path in the init script
https://github.com/AmazeeLabs/silverback-template/blob/c3c2dd959c9ad1382604cca1d028373f0a7ff02c/INIT.md#L99-L103

BTW, auth-key a good candidate for the central env var management.

File: tests/e2e/webform-snapshots/webforms.spec.ts [hunk 3]
```diff
@@ -4,7 +4,7 @@ import { execSync } from 'child_process';
 
 import { cmsUrl } from '../helpers/url';
 
-const baseDir = '../../packages/ui/static/public/webforms';
+const baseDir = '../../packages/ui/static/stories/webforms';
```
Comment (Leksat) [inline tests/e2e/webform-snapshots/webforms.spec.ts:7]: Should be git-ignored?

**Review by Leksat — APPROVED** [review]
Just a few minor things and we are ready to merge 🎉 

I tried to get rid of Drupal locally very quickly. Still a challenge 😅

But this PR is a great start to the bright future 💪

**Comment by pmelab** [comment]
Thanks @Leksat ! Addressed your remarks, will merge when green! 🥳

**Pull request merged**